runtime/vm/ContinuationHelpers.cpp

```diff
@@ -112,22 +112,24 @@ synchronizeWithConcurrentGCScan(J9VMThread *currentThread, j9object_t continuati
 			internalReleaseVMAccess(currentThread);
 
 			omrthread_monitor_enter(currentThread->publicFlagsMutex);
-			while (VM_VMHelpers::isConcurrentlyScanned(continuation->state)) {
+			while (VM_VMHelpers::isConcurrentlyScanned(*continuationStatePtr)) {
```

Comment:
Let's remove this loop, but upgrade the outer 'if' to a while loop.
We gain by no checking the state while not having VM access (and dealing with potentially a moving object/its state).
We lose a bit but reacquiring VM access for each iteration of the loop, but it's really marginal loss, since this inner while loop should iterate at most twice (once for each of concurrent GC cycle type, since an object is normally scanned only once within a cycle).